ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ContractKeys.scala

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.tests
+
+import java.util.UUID
+
+import com.daml.ledger.api.testtool.infrastructure.{LedgerSession, LedgerTest, LedgerTestSuite}
+import com.daml.ledger.api.testtool.templates.{
+  Delegated,
+  Delegation,
+  MaintainerNotSignatory,
+  ShowDelegated,
+  TextKey,
+  TextKeyOperations
+}
+import io.grpc.Status
+
+final class ContractKeys(session: LedgerSession) extends LedgerTestSuite(session) {
```

Comment:
Do you think it would be worth adding a few comments and assertion messages? Unfortunately the default assertion message is very vague. Consider though that now the framework should be able to detect the line where the test fails, so that might help already.

Reply:
Addressed by 3aa8ae2 .